### PR TITLE
Fix typos in some docs 

### PR DESCRIPTION
**Pull request by luzpaz**
Found via `codespell -q 3 -S ./packages,./commonTools/gtest`

### CODE REVIEW

File: doc/build_ref/TrilinosBuildReferenceTemplate.rst [hunk 1]
```diff
@@ -289,7 +289,7 @@ This will do the following:
 
 * Generate wrappers ``build_stats_<op>_wrapper.sh`` for C, C++, and Fortran
   (and for static builds also ``ar``, ``randlib`` and ``ld``) in the build
-  tree that will compute statics as a byproduct of every invocation of these
+  tree that will compute statistics as a byproduct of every invocation of these
```
Comment (luzpaz) [inline doc/build_ref/TrilinosBuildReferenceTemplate.rst:292]: I'll revert this change.

Comment (luzpaz) [reply doc/build_ref/TrilinosBuildReferenceTemplate.rst]: Reverted

**Review by cgcgcg — APPROVED** [review]
lgtm

**Comment by cgcgcg** [comment]
Thanks, @luzpaz !

**Pull request merged**